proto/viam/app/data/v1/data.proto

```diff
@@ -47,6 +47,22 @@ enum Order {
   ORDER_ASCENDING = 2;
 }
 
+enum Status {
+  STATUS_UNSPECIFIED = 0;
+  STATUS_FAILURE = 1;
+  STATUS_PARTIAL_SUCCESS = 2;
+  STATUS_SUCCESS = 3;
+}
+
+message Error {
+  string message = 1;
```

Comment:
I think if we want to use this error to log all of the errors we encounter in the partial success scenario, we would want this to be `repeated string messages`.

Reply:
I was thinking it was be wrapped/flattened into one error message. It'll be easier to process so we don't have to loop. Not sure if there would be a big difference on how the user sees it though. Maybe @agreenb can comment??

Reply:
I think either way would probably be fine. I would maybe just add a comment to clarify that it is an aggregated error instead of a single error, like we usually expect.

Reply:
It works to aggregate but document accordingly. To simplify, can be:

```
message Result {
  Status status = 1;
  string message = 2;
}
```